### PR TITLE
Fix/remove merge accounts

### PR DESCRIPTION
**Pull request by alymdrictels**
This PR adds safety checks for field and image mappings, removes the need to manually add doctrine mappings to the configuration, removes the need for default parameters in a yml (the bundle Extension handles that) and removes support for the 'merge_accounts' feature.

### CODE REVIEW

File: DependencyInjection/NetgenEzSocialConnectExtension.php [hunk 1]
```diff
@@ -62,13 +84,66 @@ public function getAlias()
...
+        foreach ($extensionConfig['resource_owners'] as $resourceOwnerName => $resourceOwnerValues)
+        {
+            $this->hwiFallbackData[$resourceOwnerName] = array();
+
+            // Do nothing if hwi configuration has dynamic parameters (i.e. we are using siteaccess-aware configuration)
+            if (mb_strpos('%', $resourceOwnerValues['client_id']) === false) {
+                $this->hwiFallbackData[$resourceOwnerName]['id'] = $resourceOwnerValues['client_id'];
+            }
+
+            if (mb_strpos('%', $resourceOwnerValues['client_secret']) === false) {
```
Comment (iherak) [inline DependencyInjection/NetgenEzSocialConnectExtension.php]: What if `client_secret` (or `client_id`) contain "%"?

File: Helper/UserContentHelper.php [hunk 2]
```diff
@@ -223,28 +253,50 @@ function (Repository $repository) use ($userCreateStruct, $userGroupId) {
...
+                }
             }
 
             $this->getRepository()->sudo(
                 function (Repository $repository) use ($user, $userUpdateStruct) {
                     return $repository->getUserService()->updateUser($user, $userUpdateStruct);
                 }
             );
+
+        } catch (\eZ\Publish\API\Repository\Exceptions\UnauthorizedException $e) {
```
Comment (iherak) [inline Helper/UserContentHelper.php]: We are using `sudo`, how could this happen? :)

File: Helper/UserContentHelper.php [hunk 3]
```diff
@@ -223,28 +253,50 @@ function (Repository $repository) use ($userCreateStruct, $userGroupId) {
...
+            $message = 'User not allowed to update the user object.';
+
+        } catch (\eZ\Publish\API\Repository\Exceptions\ContentFieldValidationException $e) {
+            $message = 'The field input is not valid.';
+
+        } catch (\eZ\Publish\API\Repository\Exceptions\ContentValidationException $e) {
+            $message = 'Required field is empty.';
+
+        } catch (\eZ\Publish\API\Repository\Exceptions\InvalidArgumentException $e) {
+            $message = 'Field type does not accept this value.';
```
Comment (iherak) [inline Helper/UserContentHelper.php]: What's the purpose of all these custom messages?

File: Helper/UserContentHelper.php [hunk 4]
```diff
@@ -373,21 +425,45 @@ public function getUserCreateStruct(OAuthEzUser $oauthEzUser, $contentType, $lan
...
+                $userCreateStruct->setField($fieldDefinitionIdentifier, $value);
+            }
+
+            return $userCreateStruct;
+        }
+
+        if ($this->logger instanceof LoggerInterface) {
+            $this->logger->error("SocialConnect: Could not map  {$fieldDefinitionIdentifier} to user content. Field does not exist.");
+
+            throw new AuthenticationException();
```
Comment (iherak) [inline Helper/UserContentHelper.php]: Shouldn't this be something like `InvalidArgumentException`?

File: Helper/UserContentHelper.php (hunk 2 above)
Comment (alymdrictels) [reply Helper/UserContentHelper.php]: True.

**Pull request merged**